components/core/tests/test-StreamingCompression.cpp

```diff
@@ -74,6 +79,11 @@ TEST_CASE("StreamingCompression", "[StreamingCompression]") {
     file_writer.close();
 
     // Decompress and compare
+    if (nullptr == decompressor) {
```

Comment:
nit: is it possible to explicilty check if the section is LZMA compression, or if the compressor is of type clp::streaming_compression::lzma::Compressor?

Reply:
I've provided a solution, but dynamic casting unique ptr is a bit cumbersome. Let me know if you like the updated solution

Reply:
https://stackoverflow.com/questions/26377430/how-to-perform-a-dynamic-cast-with-a-unique-ptr

```
//dynamic_pointer_cast overload for std::unique_ptr
template<class T, class U> std::unique_ptr<T> dynamic_pointer_cast( std::unique_ptr<U> && r ) BOOST_SP_NOEXCEPT
{
    (void) dynamic_cast< T* >( static_cast< U* >( 0 ) );

    BOOST_STATIC_ASSERT_MSG( boost::has_virtual_destructor<T>::value, "The target of dynamic_pointer_cast must have a virtual destructor." );

    T * p = dynamic_cast<T*>( r.get() );
    if( p ) r.release();
    return std::unique_ptr<T>( p );
}
```

Reply:
One concern on your code is the std::move. People may not expect that the compressor type check would invalidate the compressor object.
I assume compressor is not used afterwards, but if someone tries to use compressor in later change he/she may run into unexpected behavior. 

@kirkrodrigues 
Can you leave feedback on if this my previous comment is an overkill? What Bill recently modified is [here](https://github.com/y-scope/clp/pull/614/files#diff-be71344840baa445f34e0ef9d3f38dff2b4628024e7f48f9d40e49058a589bb3R85).

Reply:
How about extracting compression and decompression & comparison into two helpers (which is how we should write it anyway to keep functions short) and then you can call both helpers for the Zstd & Passthrough cases, and call only the first one for the LZMA case?

Reply:
Done